README.md

```diff
@@ -1,16 +1,32 @@
+## Pre-requisites
+
+- You will need to have an Azure account already available
+- Azure CLI installed to use this provider.
+- An Azure OpenAI endpoint provisioned with the models you would like to use.
+
+You will need to provide the following information when first setting up the provider:
+
+- Azure Subscription ID
+- Azure Resource Group
+- Azure OpenAI model name
+
 ## Usage Example
 
+To override the default model of all your gptscripts you can run the following command:
+
 ```
 gptscript --default-model='gpt-4o from github.com/gptscript-ai/azure-openai-provider' examples/helloworld.gpt
 ```
 
+### Troubleshooting
 
-#### Advanced
+#### I'm getting 429 errors due to rate limiting
 
-If you want to bypass the need for the Azure CLI or authentication to Azure in general, you can set `GPTSCRIPT_AZURE_ENDPOINT`, `GPTSCRIPT_AZURE_API_KEY`, `GPTSCRIPT_AZURE_DEPLOYMENT_NAME` as environment variables before using the provider for the first time. Note that these values will still be saved as a GPTScript credential, so changing the environment variables later will not effect GPTScript until you delete that saved credential.
```

Comment:
This info is confusing because of the credential system, but it is in here because a bunch of people have been interested in bypassing the requirement for needing to have the azure cli or azure credentials in general. I don't think we should remove it.

Reply:
I added it back, but I don't think this works. You need to pass in the env var overrides I think, which is terrible :)

Reply:
Yeah its a little confusing with the credential provider stuff. Ideally we could have some way of creating credentials ahead of runtime with `gptscript credential` that would avoid the need for doing it this way if the user already knows what they want.